java/flight/src/main/java/org/apache/arrow/flight/FlightStatusCode.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.flight;
+
+/**
+ * A status code describing the result of a Flight call.
+ */
+public enum FlightStatusCode {
+  OK,
+  UNKNOWN,
+  INTERNAL,
+  INVALID_ARGUMENT,
+  TIMED_OUT,
+  CANCELLED,
+  UNAUTHENTICATED,
+  UNAUTHORIZED,
+  UNIMPLEMENTED,
+  UNAVAILABLE,
+  ;
+
+  /**
+   * Create a blank {@link CallStatus} with this code.
+   */
+  public CallStatus toStatus() {
+    return new CallStatus(this, null, null);
```

Comment:
please document what the null parameter represent.

Reply:
I ended up creating the single-parameter constructor as suggested below.